ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs

```diff
@@ -1552,8 +1552,8 @@ data EffectiveDataFlow
     deriving (Eq, Show)
 
 -- | Property wrapping `multinodeExperiment`.
-prop_multinode_Sim :: Int -> ArbDataFlow -> MultiNodeScript Int TestAddr -> Property
-prop_multinode_Sim serverAcc (ArbDataFlow dataFlow) script =
+prop_multinode_Sim :: Int -> ArbDataFlow -> AbsBearerInfo -> MultiNodeScript Int TestAddr -> Property
```

Comment:
If I generate a list than the test fails calling the partial function `last` and `unidirectional_Sim` also fails. I'm not sure if a non-empty list is the best as for the tests to pass I believe we should have the last `BearerInfo` used to be `noAttenuation`

Reply:
> If I generate a list than the test fails calling the partial function last and unidirectional_Sim also fails.

Let's try to fix this.

> I'm not sure if a non-empty list is the best as for the tests to pass I believe we should have the last BearerInfo used to be noAttenuation

That's right. It's common to have such invariants inside generators, i.e. generate non empty lists which last element is `noAttenuation`.  One needs to make sure that shrinking preserves the invariant (which in this case is easy to do).

Reply:
What I meant by last used was exactly the last one used, not necessarily the last one in the list, does this make sense?

Reply:
Ah, I see. That's ok then as it would need to redesign the test.   The diffusion simulation will not need to relay on such aninvariant; the outbound governor is the component which makes sure that a progress is made.  Maybe just leave a note why we do this this way, mention the limitation of the test (there's nothing that will guarantee progress like outbound governor).